File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task51_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.Scanner;

public class CaesarCipher {
    public static String encrypt(String text, int shift) {
        StringBuilder result = new StringBuilder();
        for (char character : text.toCharArray()) {
            if (Character.isLetter(character)) {
                char base = Character.isUpperCase(character) ? \'A\' : \'a\';
                result.append((char) ((character - base + shift) % 26 + base));
            } else {
                result.append(character);
            }
        }
        return result.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a string to encrypt: ");
        String text = scanner.nextLine();
        System.out.print("Enter the shift value: ");
        int shift = scanner.nextInt();
        String encrypted = encrypt(text, shift);
        System.out.println("Encrypted string: " + encrypted);
        scanner.close();
    }
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task51_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <string>

std::string encrypt(std::string text, int shift) {
    for (char &c : text) {
        if (std::isalpha(c)) {
            char base = std::isupper(c) ? \'A\' : \'a\';
            c = (c - base + shift) % 26 + base;
        }
    }
    return text;
}

int main() {
    std::string text;
    int shift;

    std::cout << "Enter a string to encrypt: ";
    std::getline(std::cin, text);

    std::cout << "Enter the shift value: ";
    std::cin >> shift;

    std::string encrypted = encrypt(text, shift);
    std::cout << "Encrypted string: " << encrypted << std::endl;

    return 0;
}
